test-util/src/main/java/io/camunda/zeebe/test/util/jmh/JMHTestCase.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.test.util.jmh;
+
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.regex.Pattern;
+import org.agrona.LangUtil;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.ChainedOptionsBuilder;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+/**
+ * A small utility to help you build and run JMH benchmarks as JUnit tests. Use with {@link
+ * JMHAssert} to assert the results.
+ *
+ * <p>You can reuse this in your JUnit tests:
+ *
+ * <pre>{@code
+ * &#64;Test
+ * void shouldBenchmark() {
+ *   // given
+ *   final var testCase = JMHTestCase.of(MyBenchmarkClass.class, "measurePropertyFoo");
+ *
+ *   // when
+ *   final var assertResult = testCase.run();
+ *
+ *   // then
+ *   assertResult.isWithinDeviation(3.01, 0.2);
+ * }
+ * }</pre>
+ */
+public final class JMHTestCase {
+  private final ChainedOptionsBuilder builder;
+
+  public JMHTestCase(final ChainedOptionsBuilder builder) {
```

Comment:
🔧 I guess we can make the ctor private then

Reply:
I kept it up in case someone wanted to pass some weirdly custom thing, but I guess it can be customized via `withOptions` anyway, so :+1: